variants/deploy_with_capistrano/template.rb

```diff
@@ -0,0 +1,179 @@
+  }
+
+  # Default value for :linked_files is []
+  append :linked_files, ".env"
+
+  # Default value for linked_dirs is []
+  append :linked_dirs, "log", "tmp/pids", "tmp/cache", "tmp/sockets", "public/system", "storage"
+
+  ##
+  # Roles
+  #
+  # We have a single :app role which contains all the servers we care about in
+  # this deployment.
+  #
+  # We run nginx, rails on the same servers so we do not need the distinction
+  # between the :web and :app roles that Capistrano encourages by default.
+  #
+  # Capistrano is not involved at all in Database management so we don't need a
+  # separate :db role either
+  #
+  set :assets_roles, [:app] # defaults to [:web]
+  set :migration_role, :app # defaults to :db
+
+  # Default value for default_env is {}
+  set :default_env, path: "$HOME/.rbenv/shims:$HOME/bin:/snap/bin:$PATH"
+
+  set :rbenv_path, "$HOME/.rbenv"
+  set :rbenv_ruby, File.read(".ruby-version").strip
+  set :rbenv_prefix, "$HOME/.rbenv/bin/rbenv exec"
+  set :rbenv_map_bins, %w[rake gem bundle ruby rails]
+
+  namespace :deploy do
+    desc "Restart application"
+    task :restart do
+      on roles(:app), in: :sequence, wait: 5 do
+        # TODO: replace these examples with the real commands that control your
+        # server
+        execute :sudo, "systemctl restart puma"
+      end
+    end
+
+    task :start do
+      on roles(:app) do
+        execute :sudo, "systemctl start puma"
+      end
+    end
+
+    task :stop do
+      on roles(:app) do
+        execute :sudo, "systemctl stop puma"
+      end
+    end
+
+    task :status do
+      on roles(:app) do
+        execute :sudo, "systemctl status puma"
+      end
+    end
+
+    after :publishing, :restart
+  end
+
+EO_RUBY
+
+gsub_file("config/deploy.rb", old_generated_cap_config_snippet, new_ackama_cap_config_snippet)
+
+insert_into_file "Capfile", after: /install_plugin Capistrano::SCM::Git/ do
+  <<~'EO_RUBY'
+    # Include tasks from other gems included in your Gemfile
+    #
+    # For documentation on these, see for example:
+    #
+    #   https://github.com/capistrano/rbenv
+    #   https://github.com/capistrano/bundler
+    #   https://github.com/capistrano/rails
+    #   https://github.com/sheharyarn/capistrano-rake
+    #
+    require "capistrano/rbenv"
+    require "capistrano/bundler"
+    require "capistrano/rails/assets"
+    require "capistrano/rails/migrations"
+    require "capistrano/rake"
+  EO_RUBY
+end
+
+# Example:
+# deploy_envs = {"production"=>"config/deploy/production.rb", "staging"=>"config/deploy/staging.rb"}
+deploy_envs = Dir.children("config/deploy")
+                 .each_with_object({}) do |file_name, acc|
+  key = File.basename(file_name, ".rb")
+  acc[key] = "config/deploy/#{file_name}"
+end
+
+deploy_envs.each do |env_name, file_path|
+  prepend_to_file(file_path) do
+    <<~EO_RUBY
+      # These are the most common settings required to deploy to a server
+      set :rails_env, "#{env_name}"
+      set :branch, "#{env_name}"
```

Comment:
This will be incorrect for at least one branch, because we have `main` deploy to either staging or uat - I think it would be best to assume `main` -> `staging` because 1. that might push us slightly more towards being consistent in our env names and 2. having `staging` as the branch pushes people towards the wrong default (i.e. I think its more likely someone will assume they should create a branch named `staging` than to change this line to be `main`)